### DISCUSSION TITLE
TTS concurrency limit?

### DISCUSSION THREAD

**Discussion by jasonm** [deepgram/community#791] [General help] [answered]
[The documented rate limit for TTS is 480 req/min](https://developers.deepgram.com/reference/api-rate-limits#text-to-speech) but I can also produce HTTP 429 responses with a small number (<10) of concurrent requests. We are building an STT/LLM/TTS using Deepgram for STT and TTS (nova-2), so we do not need more than a few concurrent requests per user (just for pipelining for latency reduction) but how does this concurrency limit apply across multiple concurrent users (on different machines, IPs, etc.)? Is there some concurrent limit applied across all of them, eg per Deepgram API key? Thank you!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @jasonm, our rate limits are applied across an entire Deepgram project, no matter which API keys are used. 

We enforce TTS rate limits of 2 concurrent requests. Our average latency per request is 250 ms, so we extrapolate that to 8 requests per second, or 480 requests per minute. That's why you would observe 429s even while being far below 480 concurrent requests.

Depending on your use case, you could space out your requests more, and/or set up retries with exponential backoff, to ensure that requests that do get rate-limited eventually succeed.

We also have a [related blog post](https://deepgram.com/learn/api-back-off-strategies) on backoff strategies for Deepgram rate limiting.

**Reply by lingxuan0520** [reply]
This seems to be the logic for transcribing audio files. I understand it.If it is the nova-3 streaming voice-to-text service, it supports a maximum of 50 concurrent connections. Does this mean that 50 users can simultaneously use the same API at the same time?or what is the logic for streaming stt ? Regards

**Reply by jkroll-deepgram** [reply]
Hi @lingxuan0520, the rate limit of 50 concurrent connections for streaming STT is concurrent websocket connections. This typically corresponds to number of concurrent calls. So yes, 50 end-users can simultaneously use streaming STT at the same time.
